specification.md

```diff
@@ -2543,12 +2565,20 @@ workflowId: handleApprovedVisaWorkflowID
 
 </details>
 
+The `workflowId` property defined the unique ID of the sub-workflow to be invoked.
+
+The `version` property defined the unique version of the sub-workflow to be invoked.
+If this property is defined, runtimes should match both the `id` and the `version` properties
+defined in the sub-workflow definition. If it is not defined, the lookup should be done by
+`workflowId` property alone. In the case of multiple sub-workflows definitions that match this
+lookup, runtimes should report this as errors.
```

Comment:
Nit; can we say that the runtime should define the behavior when the version is ambiguous?  For example, some might have a concept of the latest or 'production' version of a workflow and default to that when no version is specified?

Reply:
I would still prefer to make it explicit as will make it more portable in my opinion. Leaving the version resolution to the runtimes is fine but still the version should be specified and it could be like meta version as you said (`latest` etc)

The spec could mention use of `latest` or `default` versions but described that they should be explicitly defined in the workflow

Reply:
@jorgenj @mswiderski Shouldn't the version, if not specified, default to latest? That's the actual behavior of my runtime. Using 'default or  'latest seems fine to me too, but having this field as required doesn't feel totally right, especially if I compare to, say, Docker or K8s image tags...

Reply:
in general I'd be fine with it as long as it goes by the spec meaning spec defines version on subflow as latest if not given so it's explicit. Rather if not set it is runtime specific as it makes it less portable. what do you think?

Reply:
@mswiderski Yes, you are right, a a default explicit value is IMHO the way to go. It leaves the possibility of not setting anything on legacy subflows, though guaranteeing consistant behavior accross runtimes.

Reply:
It seems that `latest` might be a potentially dangerous choice.  Consider the case where devs are uploading new versions of their workflow for testing, if they forgot to specify a version in the parent workflow, then their 'production' workflows might start spawning 'dev' (unapproved) versions of the subflow.

I think `default` is reasonable, but I'm not sure exactly how `default`'s behavior would be defined at this point other than to say it can be runtime-specific (unless this spec is now going to get into the topic of managing versions, like 'approving' a new version to be the `default` or something like that).

Reply:
@mswiderski @Neuroglia @jorgenj 
I do think that defining "latest" covers us from the spec perspective but as @jorgenj mentioned that could lead to issues due to possible mistakes. Looking at it only from the spec perspective it does seem adding a default value of "latest" is fine for portability.

Reply:
agreed. Potential user mistakes are unavoidable anyway and I don't think the spec can do much about it. Enforcing version (either to be explicitly given or defined as default) should be enough in my opinion.

Reply:
Avoiding user mistakes is one thing, but making the default behavior something inherently unsafe like 'latest' (use whatever happens to have been uploaded last for production workflows?) is just guaranteeing that many users will make that mistake.

Reply:
isn't that exactly what container images do by default (as pointed by @Neuroglia)? If you don't specify the tag it assumes `latest` .I think people are already used to this behavior and once spec clearly states it there should not be any issues with that.

Reply:
In my experience the generally accepted best-practice is "don't use latest for deployed code" [[1](https://vsupalov.com/docker-latest-tag/)] 

From the [docker blog on best practices](https://docs.docker.com/develop/dev-best-practices/):

> When building images, always tag them with useful tags which codify version information, intended destination (prod or test, for instance), stability, or other information that is useful when deploying the application in different environments. Do not rely on the automatically-created latest tag.

We can also look to Maven's snapshot feature as a case-study with similar results, as far as I'm aware the general advice is "don't use SNAPSHOT for deployed code".

Reply:
I agree that the latest version should not be considered to be used for production but it does make sense to have it in the same way it is for container images - recommended to use concrete versions. So having it defined by the spec will make runtimes more aligned.